### PR TITLE
Icon refactor

### LINKED ISSUE #1749: Icon base component refactor
- [ ] Drop `pullRight` prop
  Make sure all usages of this prop still display properly using CSS in parent component
- [ ] Make `handleClick` function into a React hook using `useCallback`
- [ ] Replace `bsStyle` with `color` prop (eg. red, green, orange)
  Get colors from Styled Components theme

### PR DESCRIPTION
**Pull request by jinxuchen**
- resolve #1749 
- refactored Icon base component
- update tests

### CODE REVIEW

File: client/src/js/base/__tests__/Icon.test.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { Tooltip } from "../Tooltip";
 import { Icon } from "../Icon";
+import { StyledIcon } from "../Icon";
```
Comment (igboyes) [inline client/src/js/base/__tests__/Icon.test.js]: Codacy Issue found: ['../Icon' import is duplicated.](https://app.codacy.com/manual/virtool/virtool/pullRequest?prid=5219009)

**Comment by igboyes** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/virtool/virtool/pullRequest?prid=5219009&bid=17110669)

**Comment by igboyes** [comment]
Do you want me to fix this conflict?

**Comment by igboyes** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/virtool/virtool/pullRequest?prid=5219009&bid=17110669)

File: client/src/js/base/__tests__/Icon.test.js (hunk 1 above)
Comment (igboyes) [inline client/src/js/base/__tests__/Icon.test.js]: Codacy Issue found: ['../Icon' import is duplicated.](https://app.codacy.com/manual/virtool/virtool/pullRequest?prid=5219009)

**Comment by jinxuchen** [comment]
> Do you want me to fix this conflict?

yes please.

**Pull request merged**